changelog/transition-complex.dd

```diff
@@ -0,0 +1,27 @@
+Complex and imaginary types are now deprecated
+
+D currently supports complex and imaginary versions of all floating point types.
+---
+float a = 2;
+ifloat b = 4i;
+cfloat c = a + b;
+assert(c == 2 + 4i);
+---
+
+These types are too specialized to be a part of the core language, and have
+been transitioned over to library types in `std.complex`.
+
+The transition phase for complex and imaginary types has finished, and
+deprecation warnings are now turned on by default.
```

Comment:
```suggestion
However these types are too specialized to be a part of the core language,
and the same functionalities can be implemented as part of a library type.
As such, older versions of DMD provided the `-transition=complex` switch to
warn when those builtin types were used.
This transition phase has finished, and those deprecations are now turned on by default.
```

The sentences before felt a bit disconnected, especially between "These types [...]" and "The transition phase [..]", hence why I mention the `-transition` switch here rather than at the end. Feel free to reword if you have some better wording in mind.